### ISSUE TITLE
require-prop-comment

### ISSUE THREAD

**Issue by czb3279338858** [vuejs/eslint-plugin-vue#1993]
**Please describe what the rule should do:**

require prop should have a comment

**What category should the rule belong to?**

[ ] Enforces code style (layout)
[ ] Warns about a potential error (problem)
[X ] Suggests an alternate way of doing something (suggestion)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule should warn about:**

```vue
      <template>
        <div>1</div>
      </template>
      <script setup>
      const props = defineProps({
        a: Number
      })
      </script>
```
```vue
      <template>
        <div>1</div>
      </template>
      <script>
      import { defineComponent } from '@vue/composition-api'

      export default defineComponent({
        props: {
          /**
           * a comment
           */
          a: Number,
          b: Number,
          // c
          // comment
          c: Number
        },
        setup() {}
      })
      </script>
```

**Additional context**

**Comment by FloEdelmann** [comment]
I like the rule name and can see the benefit of adding this rule.
